pymatgen/io/lobster/inputs.py

```diff
@@ -301,27 +271,53 @@ def from_dict(cls, dct: dict) -> Self:
         Returns:
             Lobsterin
         """
-        return cls({k: v for k, v in dct.items() if k not in ["@module", "@class"]})
+        return cls({key: val for key, val in dct.items() if key not in {"@module", "@class"}})
+
+    def _get_nbands(self, structure: Structure) -> int:
+        """Get number of bands."""
+        if self.get("basisfunctions") is None:
+            raise ValueError("No basis functions are provided. The program cannot calculate nbands.")
+
+        basis_functions: list[str] = []
+        for string_basis in self["basisfunctions"]:
+            string_basis_raw = string_basis.strip().split(" ")
+            while "" in string_basis_raw:
+                string_basis_raw.remove("")
+            for _idx in range(int(structure.composition.element_composition[string_basis_raw[0]])):
+                basis_functions.extend(string_basis_raw[1:])
+
+        num_basis_functions = 0
+        for basis in basis_functions:
+            if "s" in basis:
+                num_basis_functions += 1
+            elif "p" in basis:
+                num_basis_functions += 3
+            elif "d" in basis:
+                num_basis_functions += 5
+            elif "f" in basis:
+                num_basis_functions += 7
+
+        return int(num_basis_functions)
 
     def write_INCAR(
         self,
-        incar_input: str = "INCAR",
-        incar_output: str = "INCAR.lobster",
-        poscar_input: str = "POSCAR",
-        isym: int = -1,
+        incar_input: PathLike = "INCAR",
+        incar_output: PathLike = "INCAR.lobster",
+        poscar_input: PathLike = "POSCAR",
+        isym: Literal[-1, 0] = -1,
         further_settings: dict | None = None,
-    ):
-        """Will only make the run static, insert nbands, make ISYM=-1, set LWAVE=True and write a new INCAR.
-        You have to check for the rest.
+    ) -> None:
+        """Write INCAR file. Will only make the run static,insert NBANDS,
+        set ISYM=-1, LWAVE=True and you have to check for the rest.
```

Comment:
```suggestion
        set ISYM=0, LWAVE=True and you have to check for the rest.
```

Reply:
If we were to update the default (recommended) `ISYM` value from `-1` to `0`, should we consider updating the default argument to `0` as well for consistency (`isym: Literal[-1, 0] = -1,`)? And issue a warning for such changes in behavior?

Reply:
Yes, it would be good to do this.

Reply:
> Yes, it would be good to do this.

I made this change in b351443aaf521b3b1fe729f288c0a6f6e91259cc.

> The effect for the user should be minimal as LOBSTER and VASP run for both settings.

I think this is even more dangerous than a breaking change, because user might not even notice the default value has changed. So I set the default to `None` to detect if the default value is being used, instead of directly changing `-1` to `0`.

Reply:
I would set the default value to 0. The only difference between 0 and -1 is how long the nonSCF-VASP and LOBSTER calculations take. A user really needing -1 will be aware of it.